flaskbb/plugins/models.py

```diff
@@ -129,15 +141,51 @@ def add_settings(self, settings, force=False):
 
             pluginstore.key = key
             pluginstore.plugin = self
-            pluginstore.value = settings[key]['value']
-            pluginstore.value_type = settings[key]['value_type']
-            pluginstore.extra = settings[key]['extra']
-            pluginstore.name = settings[key]['name']
-            pluginstore.description = settings[key]['description']
+            pluginstore.value = settings[key]["value"]
+            pluginstore.value_type = settings[key]["value_type"]
+            pluginstore.extra = settings[key]["extra"]
+            pluginstore.name = settings[key]["name"]
+            pluginstore.description = settings[key]["description"]
             plugin_settings.append(pluginstore)
 
         db.session.add_all(plugin_settings)
         db.session.commit()
 
+    def upgrade_settings(self):
+        plugin_module = current_app.pluggy.get_plugin(self.name)
+        settings = plugin_module.SETTINGS
+
+        deleted_settings = [
+            (key, s) for key, s in self.values.items() if key not in settings
+        ]
+        added_settings = [
+            PluginStore.from_dict(self, key, values)
+            for key, values in settings.items()
+            if key not in self.values
+        ]
+        # but what do here?
+        # updated_settings = []
```

Comment:
@sh4nks did we ever come up with a plan for updated settings? Still not sure what approach is best tbh

Reply:
Nope not really, IIRC we talked about versioning the settings but I am not exactly sure how this would solve this.